src/main/java/com/exadel/discountwebapp/statistics/excelexport/ExcelExporter.java

```diff
@@ -0,0 +1,151 @@
+            createCell(row, columnCount++, pair.getKey().getEmail(), style);
+            createCell(row, columnCount, pair.getValue(), style);
+            if (pair.getKey() instanceof OthersUsersVO) {
+                createCell(row, 0, ((OthersUsersVO) pair.getKey()).getTitle(), style);
+            }
+        }
+
+        rowCount++;
+        Row rowTitleCategory = sheet.createRow(rowCount++);
+        Row rowCategory = sheet.createRow(rowCount++);
+
+        createCell(rowTitleCategory, 0, "Top most popular Categories", style16);
+        createCell(rowCategory, 0, TITLE, style15);
+        createCell(rowCategory, 1, QUANTITY, style15);
+
+
+        for (Map.Entry<CategoryVO, Long> pair : summaryStats.getPopularCategoriesStats().entrySet()) {
+            Row row = sheet.createRow(rowCount++);
+            var columnCount = 0;
+            createCell(row, columnCount++, pair.getKey().getTitle(), style);
+            createCell(row, columnCount, pair.getValue(), style);
+        }
+
+        rowCount++;
+        Row rowTitleVendor = sheet.createRow(rowCount++);
+        Row rowVendor = sheet.createRow(rowCount++);
+
+        createCell(rowTitleVendor, 0, "Top most popular Vendors", style16);
+        createCell(rowVendor, 0, TITLE, style15);
+        createCell(rowVendor, 1, QUANTITY, style15);
+
+        for (Map.Entry<VendorVO, Long> pair : summaryStats.getPopularVendorsStats().entrySet()) {
+            Row row = sheet.createRow(rowCount++);
+            var columnCount = 0;
+            createCell(row, columnCount++, pair.getKey().getTitle(), style);
+            createCell(row, columnCount, pair.getValue(), style);
+        }
+
+        rowCount++;
+        Row rowTitleDiscountViews = sheet.createRow(rowCount++);
+        Row rowDiscountViews = sheet.createRow(rowCount++);
+
+        createCell(rowTitleDiscountViews, 0, "Top most popular discounts by views", style16);
+        createCell(rowDiscountViews, 0, "id", style15);
+        createCell(rowDiscountViews, 1, TITLE, style15);
+        createCell(rowDiscountViews, 2, QUANTITY, style15);
+
+        for (Map.Entry<DiscountVO, Long> pair : summaryStats.getPopularDiscountsStats().entrySet()) {
+            Row row = sheet.createRow(rowCount++);
+            var columnCount = 0;
+            createCell(row, columnCount++, pair.getKey().getId(), style);
+            createCell(row, columnCount++, pair.getKey().getTitle(), style);
+            createCell(row, columnCount, pair.getValue(), style);
+            if (pair.getKey() instanceof OthersDiscountsVO) {
+                createCell(row, 0, pair.getKey().getTitle(), style);
+            }
+        }
+    }
+
+    public void export(HttpServletResponse response) throws IOException {
+        writeHeaderLine();
+        writeDataLines();
+        ServletOutputStream outputStream = response.getOutputStream();
+        workbook.write(outputStream);
+        workbook.close();
+        outputStream.close();
```

Comment:
exporter shouldn't aware of closing this stream

Reply:
fixed